exporter/sumologicexporter/exporter_test.go

```diff
@@ -283,7 +284,8 @@ func TestPushFailedBatch(t *testing.T) {
 	log := logs.ResourceLogs().At(0)
 
 	for i := 0; i < maxBufferSize; i++ {
-		logs.ResourceLogs().Append(log)
```

Comment:
Is it me, or is the removal of Append going to complicate the API for the consumers? One call is now two calls (AppendEmpty + CopyTo).

Reply:
We changed to this to not change too much in components, but normally consumers should call AppendEmpty then fill the result instead of returning a metric.

Reply:
Not sure I understand. Previously, a consumer would call:

```go
			logs.Append(log)
```

Now, they need to do this:
```go
			tgt := logs.AppendEmpty()
			log.CopyTo(tgt)
```

This does complicate the caller, no?

Reply:
Actually just in this case, Append was not correctly used. Because Append does not do a Copy of the log, so all the entries would point to the same log internal (a.k.a if you change one of the entry it changes all).

But in general the previous use-case was:
```
log := generateLog()
logs.Append(log)
```
Now customers will do:
```
log := fillLog(logs.AppendEmpty())
```

Reply:
I guess this one was also used incorrectly then? 

From:
```go
				s.Events().Append(event)
```

To:
```go
				tgt := s.Events().AppendEmpty()
				event.CopyTo(tgt)
```

If so, the change looks good to me.

Reply:
It was, but to limit the amount of changes we went the simple path.